packages/comms/src/ecl/topology.ts

```diff
@@ -89,7 +89,7 @@ export class Topology extends StateObject<TopologyStateEx, TopologyStateEx> impl
             this.set({
                 Services: response.ServiceList
             });
-            return this.Services;
+            return this.Services!;
```

Comment:
I probably just don't understand what the use of this operator is, generally. But does this not clash with the `| undefined` that was added to the accessor? Implying it's potentially undefined in one place and then asserting it would never be undefined here.

Reply:
Its asserting that `this.Services` exists at this point in the code.  In this case we have just updated it in the previous 3 lines (89 -> 91).